_posts/2020-12-20-azure-event-grid.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: post
+current: post
+cover: 'assets/images/post-cover/20201118-cloudconf.jpg'
+socialcover: 'assets/images/post-cover/20201118-cloudconf-s.jpg'
+slug: 'cloud-conf'
+title: 'Azure Event Grid: a message to rule them all'
+date: 2020-11-18 09:00:00
+category : tech
+tags: [azure, event grid, distributed systems, reactive systems]
+author: Ale
+---
+Immaginiamo di vivere in un mondo parallelo in cui, a fronte di ogni azione, venga generato un evento. E immaginiamo che esista una piattaforma/servizio che nativamente metta a disposizione queste caratteristiche. Sarebbe fantastico, no? Beh, forse una soluzione gia' c'e'...Azure Event Grid
+
+[Azure Event Grid](https://docs.microsoft.com/en-us/azure/event-grid/?WT.mc_id=AZ-MVP-5001625) non e' l'unico servizio disponibile su Azure che ha a che fare con "la messaggistica". Prima di capire come usare Azure Event Grid e' quindi forse meglio capire quando Event Grid fa al caso nostro.
+
+Nel "mondo" dei messaggi, possiamo identificare due macro-categorie che ne definiscono l'ambito: i comandi e gli eventi.
+
+I comandi, come da [definizione](https://www.treccani.it/vocabolario/comando1/) sono assertivi: qualcuno chiede a qualcun altro di fare qualcosa.  
+La comunicazione e' "point-to-point", tra due attori del sistema e, solitamente, il produttore del comando e' interessato al risultato dell'operazione richiesta.
+
+Gli eventi, invece, vengono utilizzati per notificare che qualcosa e' avvenuto nel sistema, che il sistema ha subito un cambio di stato. La comunicazione non e' piu' tra due attori specifici, ma tra un componente e N eventuali altri componenti interessati a quel tipo di informazione. E' una comunicazion di tipo broadcast, in cui il numero di componenti N puo' essere _uno, nessuno o centomila_.
+
+Focalizzando l'attenzione sulla categoria degli eventi, possiamo affinare la distinzione individuando due peculiarita' che possono caratterizzare e distinguere gli eventi:
+- eventi "atomici": ovvero eventi che hanno la loro ragione di essere completamente scorrelata dalla produzione di altri eventi dello stesso tipo. Per esempio "un ordine e' stato registrato" in un sistema di e-commerce potrebbe essere un evento di questo tipo
+- eventi correlati (nel tempo): il contenuto informativo di tali eventi non e' rappresentato dall'i-esimo evento in particolare, ma dallo stream di tutti gli eventi. Basti pensare ad un sensore di temperatura: e' molto piu' importante sapere se la "tendenza" della temperatura e' in discesa, per attivare la caldaia, piuttosto che sapere che in un particolare istante temporale, la temperatura rilevata era di 20.3 gradi
```

Comment:
Cambierei un po' la punteggiatura: _e' molto piu' importante sapere se la "tendenza" della temperatura e' in discesa, per attivare la caldaia, piuttosto che sapere che in un particolare istante temporale la temperatura rilevata era di 20.3 gradi_ oppure _e' molto piu' importante sapere se la "tendenza" della temperatura e' in discesa, per attivare la caldaia, piuttosto che sapere che, in un particolare istante temporale, la temperatura rilevata era di 20.3 gradi_